Source/WebCore/rendering/RenderLayerCompositor.cpp

```diff
@@ -3940,6 +3946,11 @@ bool RenderLayerCompositor::requiresCompositingForOverflowScrolling(const Render
     return layer.hasCompositedScrollableOverflow();
 }
 
+bool RenderLayerCompositor::requiresCompositingForAnchorPositioning(const RenderLayer& layer) const
+{
+    return !!layer.snapshottedScrollOffsetForAnchorPositioning();
```

Comment:
`has_value()`?

Reply:
why?

Reply:
Is the `!!` even necessary here? From https://en.cppreference.com/w/cpp/utility/optional:
> When an object of type optional<T> is [contextually converted to bool](https://en.cppreference.com/w/cpp/language/implicit_conversion#Contextual_conversions), the conversion returns true if the object contains a value and false if it does not contain a value.

Reply:
not really but I like to make boolean conversions explicit so they don't look like accidents

Reply:
I suppose it is not even a conversion so rather unnecessary indeed